wgpu-core/src/device/mod.rs

```diff
@@ -677,13 +694,64 @@ impl<A: HalApi> Device<A> {
             return Err(resource::CreateTextureError::InvalidMipLevelCount(mips));
         }
 
-        let raw = unsafe {
+        let raw_texture = unsafe {
             self.raw
                 .create_texture(&hal_desc)
                 .map_err(DeviceError::from)?
         };
 
-        let mut texture = self.create_texture_from_hal(raw, self_id, desc, format_features);
+        let clear_mode = if hal_usage.contains(hal::TextureUses::DEPTH_STENCIL_WRITE)
+            || hal_usage.contains(hal::TextureUses::COLOR_TARGET)
+        {
+            let (is_color, usage) =
+                if desc.format.describe().sample_type == wgt::TextureSampleType::Depth {
+                    (false, hal::TextureUses::DEPTH_STENCIL_WRITE)
+                } else {
+                    (true, hal::TextureUses::COLOR_TARGET)
+                };
+            let dimension = match desc.dimension {
+                wgt::TextureDimension::D1 => wgt::TextureViewDimension::D1,
+                wgt::TextureDimension::D2 => wgt::TextureViewDimension::D2,
+                wgt::TextureDimension::D3 => wgt::TextureViewDimension::D2, // Todo? Is this even specified?
+            };
+
+            let mut clear_views = SmallVec::new();
+            for layer in 0..desc.size.depth_or_array_layers {
+                for mip_level in 0..desc.mip_level_count {
+                    unsafe {
+                        clear_views.push(
+                            self.raw
+                                .create_texture_view(
+                                    &raw_texture,
+                                    &hal::TextureViewDescriptor {
```

Comment:
nit: can we move the creation of this descriptor out of `unsafe`? the less code in unsafe the better